src/txn/txn.c

```diff
@@ -138,6 +138,48 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
     }
 }
 
+/*
+ * __wt_txn_user_active --
+ *     Check whether there are any running user transactions. Note that a new transactions may start
+ *     on a session we have already examined and the caller needs to be aware of this limitation.
+ */
+bool
+__wt_txn_user_active(WT_SESSION_IMPL *session, bool ignore_prepare)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_SESSION_IMPL *session_in_list;
+    uint32_t i, session_cnt;
+    bool txn_active;
+
+    conn = S2C(session);
+    txn_active = false;
+
+    /*
+     * No lock is required because the session array is fixed size, but it may contain inactive
+     * entries. We must review any active session, so insert a read barrier after reading the active
+     * session count. That way, no matter what sessions come or go, we'll check the slots for all of
+     * the user sessions for active transactions when we started our check.
+     */
+    WT_ORDERED_READ(session_cnt, conn->session_cnt);
+    for (i = 0; i < session_cnt; i++) {
+        WT_STAT_CONN_INCR(session, txn_sessions_walked);
+        session_in_list = &conn->sessions[i];
+        /*
+         * Check if a user session has a running transaction. Ignore a prepared transaction if asked
+         * by the caller.
+         */
+        if (F_ISSET(session_in_list->txn, WT_TXN_RUNNING) &&
+          !F_ISSET(session_in_list, WT_SESSION_INTERNAL) &&
+          !(F_ISSET(session_in_list->txn, WT_TXN_PREPARE) && ignore_prepare)) {
```

Comment:
This function is called only one place as of now. Shall we exclude the `ignore_prepare` variable and by default exclude prepared transactions?

Reply:
I have no preference here. I can remove the `ignore_prepare` variable and indicate in method description that we are ignoring the prepared transaction.

Reply:
Fixed in 69c55ba.